src/executor_config/barrier.md

```diff
@@ -0,0 +1,89 @@
+-TODO: All Barriers Explained
+-TODO: Security Note about misconfigured barriers
+-TODO: Example
+
+# Barrier
+Before any XCMs are executed in the XCM executor, they must pass the Barrier.
+The Barrier type implements the `ShouldExecute`
+[trait](https://paritytech.github.io/polkadot/doc/xcm_executor/traits/trait.ShouldExecute.html)
+and can be viewed as the firewall for the xcm-executor. 
+Each time the xcm-executor receives an XCM, it checks with the barrier to determine if the XCM should be executed.
+
+We can also define multiple barriers for our Barrier type using a tuple.
+During execution, each barrier is checked, and if any of them succeed, the XCM is executed.
+The combination of barriers is especially important as it can either block or allow certain XCM
+functionalities. We will show an example of how barriers can be combined in the [example section]().
+
+## Implementations
+The [xcm-builder
+directory](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-builder/src/barriers.rs)
+has many implementations for the Barrier type:
+- `TakeWeightCredit`
+- `AllowTopLevelPaidExecutionFrom`
+- `AllowUnpaidExecutionFrom`
+- `AllowExplicitUnpaidExecutionFrom`
+- `IsChildSystemParachain`
+- `AllowKnownQueryResponses`
+- `AllowSubscriptionsFrom`
+- `WithComputedOrigin`
+- `TrailingSetTopicAsId`
+- `RespectSuspension`
+- `DenyThenTry`
+- `DenyReserveTransferToRelayChain`
+
+
+### `TakeWeightCredit`
+The `TakeWeightCredit`` barrier checks if the calculated weight of the XCM does not exceed a set weight limit. 
```

Comment:
It is questionable to ask: For each of these barriers, can these explanations be just part of the rust-docs? Remember, "_what can be documented in rust-docs (or generally a lower level documentation) should be documented in there_". 

I am not familiar enough to make the call, but it seems to me that these should just be rust-docs, and the high level concepts, possibly how you compose and use barriers in the broad scope should be mentioned here.

Reply:
Or possibly, if this is paraphrasing what is already in the rust-docs, I also think there is less added value in it.

Reply:
I agree with putting this explanations in rust-docs. I also think some amount of the top-level explanations and examples should be in rust-docs too.